FlowCrypt/src/main/java/com/flowcrypt/email/api/retrofit/response/model/OrgRules.kt

```diff
@@ -193,16 +197,22 @@ data class OrgRules constructor(
       throw IllegalStateException("Not a valid email $emailAddr")
     }
 
-    val disallowedDomains = disallowAttesterSearchForDomains ?: emptyList()
-    return !disallowedDomains.any { it.equals(userDomain, true) }
+    val allowedDomains = allowAttesterSearchOnlyForDomains
+    return if (allowedDomains != null) {
+      allowedDomains.any { it.equals(userDomain, true) }
+    } else {
+      val disallowedDomains = disallowAttesterSearchForDomains ?: emptyList()
+      !disallowedDomains.any { it.equals(userDomain, true) }
+    }
   }
 
   /**
    *
    * Some orgs might want to disallow lookup on attester completely
    */
   fun disallowLookupOnAttester(): Boolean {
-    return (disallowAttesterSearchForDomains ?: emptyList()).contains("*")
+    return allowAttesterSearchOnlyForDomains == null
+        && (disallowAttesterSearchForDomains ?: emptyList()).contains("*")
```

Comment:
Also add `(original condition) || (allowAttesterSearchOnlyForDomains != null && allowAttesterSearchOnlyForDomains.isEmpty)`

Reply:
> Eventually we'll remove the * method

Maybe will be better to remove `disallowLookupOnAttester` in that case. Please look at https://github.com/FlowCrypt/flowcrypt-android/pull/1831#issuecomment-1148510533